### PR TITLE
Added service injection documentation

### PR DESCRIPTION
**Pull request by MarkRedeman**

### CODE REVIEW

File: templates.md [hunk 1]
```diff
@@ -138,6 +138,16 @@ To overwrite a section entirely, you may use the `overwrite` statement:
 
 	{{-- This comment will not be in the rendered HTML --}}
 
+#### Service injection
+We can use `@inject` to resolve services from the service container.
+
+	@inject('html', 'HtmlBuilder')
+
+	<div>
+	    {{ $html->image('/assets/images/laravel.png') }}
```
Comment (taiyaei) [inline templates.md]: Looks like you switched to spaces here, tabs please

**Comment by MarkRedeman** [comment]
@taiyaei changed and rebased.

**Comment by taiyaei** [comment]
@MarkRedeman Thanks :)

File: templates.md [hunk 2]
```diff
@@ -138,6 +138,16 @@ To overwrite a section entirely, you may use the `overwrite` statement:
 
 	{{-- This comment will not be in the rendered HTML --}}
 
+#### Service injection
+We can use `@inject` to resolve services from the service container.
+
+	@inject('html', 'HtmlBuilder')
+
+	<div>
+		{{ $html->image('/assets/images/laravel.png') }}
```
Comment (GrahamCampbell) [inline templates.md]: That is the wrong blade syntax. That would result in escaped html dumped on the page.

File: templates.md [hunk 3]
```diff
@@ -138,6 +138,16 @@ To overwrite a section entirely, you may use the `overwrite` statement:
 
 	{{-- This comment will not be in the rendered HTML --}}
 
+#### Service injection
+We can use `@inject` to resolve services from the service container.
+
+	@inject('html', 'HtmlBuilder')
```
Comment (GrahamCampbell) [inline templates.md]: HtmlBuilder is not a class that's shipped with laravel by default, and is also wrong anyway - it's not the FQCN unless you've aliased it. Choose a different example please.

Comment (MarkRedeman) [inline templates.md]: Would you rather use an example that is in the Laravel framework (if so, which would be appropriate?)? I think the HtmlBuilder is a nice example as it's probably better to only inject services that really matter when working with html. (vs for example a `StatsService` that could also be included using the `with` function.

File: templates.md (hunk 2 above)
Comment (MarkRedeman) [inline templates.md]: Forgot about the new syntax. Will change it when I change the example service.

**Comment by taylorotwell** [comment]
Would probably rather use an obviously contrived example so people don't assume they can actually inject HtmlBuilder like that out of the box.

**Comment by MarkRedeman** [comment]
@taylorotwell ok, how about the `UrlGenerator`?

**Comment by GrahamCampbell** [comment]
Maybe try something that doesn't have global helper functions - ie, something someone might actually use.

**Comment by taylorotwell** [comment]
Documented.

**Pull request closed without merging**